src/librustc_resolve/error_reporting.rs

```diff
@@ -83,13 +83,20 @@ impl<'a, 'b:'a, 'c: 'b> ImportResolver<'a, 'b, 'c> {
         &mut self,
         span: Span,
         mut path: Vec<Ident>
-    ) -> Option<Vec<Ident>> {
+    ) -> Option<(Vec<Ident>, Option<String>)> {
         // Replace first ident with `crate` and check if that is valid.
         path[0].name = keywords::Crate.name();
         let result = self.resolve_path(None, &path, None, false, span, CrateLint::No);
         debug!("make_missing_crate_suggestion:  path={:?} result={:?}", path, result);
         if let PathResult::Module(..) = result {
-            Some(path)
+            Some((
+                path,
+                Some(
+                    "`use` statements changed in Rust 2018; read more at \
+                     <https://rust-lang-nursery.github.io/edition-guide/rust-2018/module-\
+                     system/path-clarity.html>".to_string()
```

Comment:
I'm unsure whether this url will change when the edition releases.

Reply:
@steveklabnik tells me that they will post a PR to change the URL to <doc.rust-lang.org/edition-guide>

Reply:
Updated the PR with this.